pennylane/transforms/core/transform_program.py

```diff
@@ -155,6 +155,16 @@ def __repr__(self):
         contents = ", ".join(f"{transform_c.transform.__name__}" for transform_c in self)
         return f"TransformProgram({contents})"
 
+    def __eq__(self, other):
+        if not isinstance(other, TransformProgram):
+            return False
+
+        return (
+            self._transform_program == other._transform_program
+            and self._argnums == other._argnums
+            and self._classical_jacobians == other._classical_jacobians
```

Comment:
Sorry to change behavior on you, but we've decided not to consider `self._argnums` and `self._classical_jacobians` when doing the comparison. 

These properties are less part of the program's fundamental representation, and more part of temporaily caching properties for re-use later.

On the plus side, this will make testing easier!